### PR TITLE
SEP-12: Do not use Host HTTP header

### PR DESCRIPTION
**Pull request by C0x41lch0x41**
It is unnecessary to use the Host HTTP header to verify the signature.
Using the expected host the callback should be sent to is enough to validate the signature.

### CODE REVIEW

**Comment by C0x41lch0x41** [comment]
@leighmcculloch thanks!

File: ecosystem/sep-0012.md [hunk 1]
```diff
@@ -527,7 +526,7 @@ It is the wallet's responsability to:
   * Key `t`: __timestamp__
   * Key `s`: __base64 signature__
 * Verify the request freshness: _current timestamp_ - __timestamp__ < few seconds (1-2 minute(s) max)
-* Extract the HTTP Header __Host__, validate that it is equal to the Wallet Host
+* The hostname hosting the server where the wallet receives the callback is labeled as __Host__
```
Comment (leighmcculloch) [inline ecosystem/sep-0012.md:529]: Could we lower case it to try and distance the label from the HTTP header?
```suggestion
* The hostname hosting the server where the wallet receives the callback is labeled as __host__
```

**Review by leighmcculloch — COMMENTED** [review]
One ask, and can you update the patch version, and add a changelog entry?

**Comment by C0x41lch0x41** [comment]
@leighmcculloch just waiting for your final review to merge this thanks.

File: ecosystem/sep-0012.md [hunk 2]
```diff
@@ -527,13 +526,13 @@ It is the wallet's responsability to:
...
 * Verify the request freshness: _current timestamp_ - __timestamp__ < few seconds (1-2 minute(s) max)
-* Extract the HTTP Header __Host__, validate that it is equal to the Wallet Host
+* The hostname hosting the server where the wallet receives the callback is labeled as __host__
 * Extract the __body__ of the request
 * Base64 decode the __base64 signature__ to get the __signature__
 * Prepare the payload to verify the signature:
   * The __timestamp__ (as a string)
   * The character `.`
-  * The __Host__
+  * The __host__
```
Comment (leighmcculloch) [inline ecosystem/sep-0012.md:535]:
```suggestion
  * The wallet host to send the callback request to
```

**Comment by C0x41lch0x41** [comment]
@leighmcculloch are you ok to merge this?
Thanks!

File: ecosystem/sep-0012.md [hunk 3]
```diff
@@ -7,7 +7,7 @@ Author: Interstellar
 Status: Active
 Created: 2018-09-11
 Updated: 2022-06-21
```
Comment (leighmcculloch) [inline ecosystem/sep-0012.md:9]: Could you update the updated date to todays date?

**Review by leighmcculloch — COMMENTED** [review]
One tweak needed, then we're good to merge.

**Comment by C0x41lch0x41** [comment]
Done thanks!

**Review by leighmcculloch — APPROVED** [review]

**Pull request merged**